github-pullrequest-plugin/src/test/java/org/jenkinsci/plugins/github/pullrequest/trigger/JobRunnerForCauseTest.java

```diff
@@ -172,4 +183,208 @@ public void configRoundTripUnsecure(Job job) throws Exception {
+        job2.save();
+
+        MatrixProject job3 = folder.createProject(MatrixProject.class, "project3");
+        job3.setDisplayName("project1 display name");
+        job3.setConcurrentBuild(true);
+        job3.getBuildersList().add(new SleepBuilder());
+        job3.setAxes(
+                new AxisList(
+                        new TextAxis("first_axis", "first_value1"),
+                        new TextAxis("second_axis", "sec_value1")
+                )
+        );
+        configRoundTripUnsecure(job3);
+        job3.save();
+
+        testAbortRunning(job1, job2, job3);
+    }
+
+
+    private <T extends TopLevelItem> void testAbortRunning(Job<?, ?> job1, Job<?, ?> job2, Job<?, ?> job3) throws Exception {
+        Jenkins jenkins = j.getInstance();
+
+        jenkins.setNumExecutors(10);
+        jenkins.save();
+
+        jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        GlobalMatrixAuthorizationStrategy matrixAuth = new GlobalMatrixAuthorizationStrategy();
+        matrixAuth.add(Jenkins.ADMINISTER, "User");
+        jenkins.setAuthorizationStrategy(matrixAuth);
+        jenkins.save();
+
+        schedule(job1, 10, "cause1_1", 0);
+        // two PR for one project in queue
+        schedule(job1, 10, "cause1_2", 0);
+        //other number for project1
+        schedule(job1, 12, "cause1_3", 0);
+
+        schedule(job2, 10, "job2", 0);
+
+        schedule(job3, 10, "cause1_3/project3", 0);
+
+        final GitHubPRTrigger gitHubPRTrigger1 = new GitHubPRTrigger("", GitHubPRTriggerMode.HEAVY_HOOKS, null);
+        gitHubPRTrigger1.setAbortRunning(true);
+        gitHubPRTrigger1.start(job1, true); // to have working polling log files
+        final JobRunnerForCause job1RunnerForCause = new JobRunnerForCause(job1, gitHubPRTrigger1);
+
+
+        final GitHubPRTrigger gitHubPRTrigger2 = new GitHubPRTrigger("", GitHubPRTriggerMode.HEAVY_HOOKS, null);
+        gitHubPRTrigger2.setAbortRunning(true);
+        gitHubPRTrigger2.start(job2, true); // to have working polling log files
+        final JobRunnerForCause job2RunnerForCause = new JobRunnerForCause(job2, gitHubPRTrigger2);
+
+        final GitHubPRTrigger gitHubPRTrigger3 = new GitHubPRTrigger("", GitHubPRTriggerMode.HEAVY_HOOKS, null);
+        gitHubPRTrigger3.setAbortRunning(true);
+        gitHubPRTrigger3.start(job3, true); // to have working polling log files
+        final JobRunnerForCause job3RunnerForCause = new JobRunnerForCause(job3, gitHubPRTrigger3);
+
+        Thread.sleep(2000);
+
+        assertThat("All runs should go to executors", jenkins.getQueue().getItems(), Matchers.<Queue.Item>emptyArray());
+
+        ACL.impersonate(Jenkins.ANONYMOUS, new Runnable() {
+            @Override
+            public void run() {
+                assertThat("Should abort job1 -> number 10", job1RunnerForCause.abortRunning(10), is(2));
+                assertThat("Should not abort more job1 -> number 10", job1RunnerForCause.abortRunning(10), is(0));
```

Comment:
@jglick Now way to know that workflow was already requested for abort. The only marker i see it interruptResult that is private.